src/main/java/tech/toparvion/analog/service/choice/ChoicesPropertiesChangesListener.java

```diff
@@ -0,0 +1,74 @@
+package tech.toparvion.analog.service.choice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Service;
+import tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.FileWatcherProvider;
+
+import javax.annotation.Nullable;
+import java.nio.file.*;
+
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+import static tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.CHOICES_AUTO_RELOAD_EXECUTOR;
+
+/**
+ * @author Polyudov
+ * @since v0.14
+ */
+@Service
+@EnableAsync
+@ConditionalOnChoicesAutoReloadEnabled
+class ChoicesPropertiesChangesListener {
+  private static final Logger log = LoggerFactory.getLogger(ChoicesPropertiesChangesListener.class);
+
+  private final RefreshEndpoint refreshEndpoint;
+  private final FileWatcherProvider fileWatcherProvider;
+
+  @Autowired
+  ChoicesPropertiesChangesListener(RefreshEndpoint refreshEndpoint,
+                                   @Nullable FileWatcherProvider fileWatcherProvider) {
+    this.refreshEndpoint = refreshEndpoint;
+    this.fileWatcherProvider = fileWatcherProvider;
+  }
+
+  @Async(CHOICES_AUTO_RELOAD_EXECUTOR)
+  @EventListener(ApplicationReadyEvent.class)
+  public void watchFile() {
+    if (fileWatcherProvider == null) {
+      return;
+    }
+    try {
+      log.info("Start watching for the choices properties file: '{}'", fileWatcherProvider.getChoicesPropertiesPath().toAbsolutePath());
```

Comment:
I've just realized recently that `toAbsolutePath()` method can be extremely misleading when applied to a local path on Windows OS if current working dir is not on the same disk as the file denoted by current 'Path' instance. In such cases, the resulting path may point to an unexisting location and would not check in any way. Thus, it's better not to rely on this method if there is no strong need for it.